src/Classes/Particle.ts

```diff
@@ -104,23 +104,25 @@ export class Particle {
         const shapeType = options.particles.shape.type;
 
         if (shapeType instanceof Array) {
-            const selectedShape = shapeType[Math.floor(Math.random() * shapeType.length)];
-
-            this.shape = selectedShape;
+            this.shape = shapeType[Math.floor(Math.random() * shapeType.length)];
         } else {
             this.shape = shapeType;
         }
 
         if (this.shape === ShapeType.image) {
             const shape = options.particles.shape;
-            this.img = {
-                ratio: shape.image.width / shape.image.height,
-                replaceColor: shape.image.replace_color,
-                src: shape.image.src,
+            const index = Math.floor(Math.random() * container.images.length);
+            const image = container.images[index];
+            const optionsImage = shape.image instanceof Array ? shape.image[index] : shape.image;
+            this.image = {
+                ratio: optionsImage.width / optionsImage.height,
+                replaceColor: optionsImage.replace_color,
+                src: optionsImage.src,
+                data: image,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The key 'data' is not sorted alphabetically](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5098130)